File: python/minicurso/exemplos_python/contar_letras.py
frase = raw_input('Digite uma frase: ')

contador = {}

for caractere in frase:
    contador[caractere] = 0

for caractere in frase:
    contador[caractere] += 1

for caractere, quantidade in contador.iteritems():
    print "Caractere %c apapreceu %d vez(es)" % (caractere, quantidade)

File: python/minicurso/exemplos-java/src/pythonjava/ContarLetras.java
package pythonjava;

import java.util.HashMap;
import java.util.Map;
import java.util.Map.Entry;

import javax.swing.JOptionPane;

public class ContarLetras {

	public static void main(String[] args) {
		String frase = JOptionPane.showInputDialog("Digite uma frase");

		Map<Character, Integer> contador = new HashMap<>();

		for (int i = 0; i < frase.length(); ++i) {
			contador.put(frase.charAt(i), 0);
		}

		for (int i = 0; i < frase.length(); ++i) {
			char c = frase.charAt(i);
			contador.put(c, contador.get(c) + 1);
		}

		for (Entry<Character, Integer> entry : contador.entrySet()) {
			System.out.printf("Caractere %c apareceu %d vez(es)\n",
					entry.getKey(), entry.getValue());
		}

	}

}
